docs/OpenTypeFeatureFileSpecification.md

```diff
@@ -1267,6 +1293,44 @@ rules, the implementation sorts the rules to avoid conflict; for example, the
 ligature substitution rule for f_f_i will be written before the ligature
 substitution rule for f_i, no matter what their order is in the feature file.
 
+<a name="4.e.i"></a>
+#### 4.e.i Feature Variations (variable font feature replacement)
+
+For variable fonts *only*, it is possible to specify [FeatureVariations](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#featurevariations-table), which allow the use of different lookups for different parts of the variation space.
+
+<a name="4.e.i.1"></a>
+##### 4.e.i.1 `conditionset`
+
+A named `conditionset` defines [conditions](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#conditionset-table) that can trigger a variable font lookup to swap in different lookups when the conditions are matched. A `conditionset` is simply a list of axis tags and min/max values. A conditionset is matched when all conditions are true. Syntax:
+```fea
+conditionset <name of set> {
+    <axis tag> <minValue> <maxValue>;
+    # additional tag min max entries
+} <name of set>;
+```
+For example, to define a "heavy" condition, for the 'wght' axis between 700-900:
+```fea
+conditionset heavy {
+    wght 700 900;
+} heavy;
+```
+
+<a name="4.e.i.2"></a>
+##### 4.e.i.2 `variation` (FeatureVariation)
+A `variation` is akin to a regular `feature` definition, with an added `conditionset` specifier to indicate that the feature should be active when the conditions of the conditionset are met. The syntax is:
+```fea
+variation <feature tag> <conditionset name or NULL> {
```

Comment:
what's the meaning of `NULL` here?

Reply:
maybe it means this (quoting from [ot spec](https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#featurevariations-table)):

> If a given condition set contains no conditions, then it matches all contexts, and the associated feature table substitution is always applied

Reply:
Right, `NULL` here is for "a condition set with no conditions". Although I suppose we could push that back and require a `conditionset` here, even if it is:
```
conditionset empty {
} empty;
```
Do you have a preference?

Reply:
What's the difference between the two following?
```
variation <feature tag> NULL {
# feature specifications
} <feature tag>;
```
and
```
feature <feature tag> {
# feature specifications
} <feature tag>;
```

Reply:
@moyogo `variation`s are very similar to `feature` definitions, but are stored differently in OT tables, and also have the possibly-NULL `conditionset` as part of their definition.